### DISCUSSION TITLE
Why caching images on disk will impact performace?

### DISCUSSION THREAD

**Discussion by iKirby** [robertodoering/harpy#708] [Q&A] [answered]
Hi, I'm new to harpy. When trying the app, I found that images are loaded from network every time after the app restarts. Then I searched here and found https://github.com/robertodoering/harpy/pull/331, it removed disk cache of images.

I agree that loading cache from disk is not as fast as loading from memory. But when the app restarts memory cache is gone and images need to be loaded from network, which is much slower than loading from disk cache and will use more network traffic.

When using a slow network, images being reloaded again and again is definitely not a good experience, and it may result in more data charges on mobile network.

**Comment by robertodoering** [comment, marked as answer]
Before #331, a lot of users reported laggy scrolling due to a low framerate and skipped frames.
Loading images from disk was a big contributor to those issues.

While I agree that fetching the same images multiple times is a waste, improving performance is more important to me.
Additionally, images in your feed will change frequently to a point where cached images might not provide a lot of value anyways.

I might experiment with just caching user profile images for people that you follow, but caching Tweet images is probably not a good idea.

**Reply by iKirby** [reply]
Thanks for your reply!
